proxy/proto/redis/proxy_conn.go

```diff
@@ -183,31 +189,57 @@ func (pc *proxyConn) Encode(m *proto.Message) (err error) {
 	if !ok {
 		return ErrBadAssert
 	}
-	switch req.mType {
-	case mergeTypeOK:
-		err = pc.mergeOK(m)
-	case mergeTypeJoin:
-		err = pc.mergeJoin(m)
-	case mergeTypeCount:
-		err = pc.mergeCount(m)
-	default:
-		if !req.IsSupport() {
-			req.reply.respType = respError
-			req.reply.data = req.reply.data[:0]
-			req.reply.data = append(req.reply.data, notSupportDataBytes...)
-		} else if req.IsCtl() {
-			reqData := req.resp.array[0].data
-			if bytes.Equal(reqData, cmdPingBytes) {
-				req.reply.respType = respString
-				req.reply.data = req.reply.data[:0]
-				req.reply.data = append(req.reply.data, pongDataBytes...)
-			} else if bytes.Equal(reqData, cmdQuitBytes) {
-				req.reply.respType = respString
+	// general supported cmd need authorized
+	if !pc.authorized && !req.IsAuth() {
```

Comment:
should not forward any command when `pc.authorized` is false. And so that authorized should be atomic flag or guard by mutex (be used by encoder/decoder together).